app/helpers/local_time_helper.rb

```diff
@@ -2,19 +2,25 @@ module LocalTimeHelper
   def local_time(time, options = nil)
     time = utc_time(time)
 
-    options, format12 = extract_options_and_value(options, :format)
-    format12, format24 = find_time_formats(format12)
+    options, format = extract_options_and_value(options, :format)
 
     options[:data] ||= {}
-    options[:data].merge! local: :time, format: format12, format24: format24
+    options[:data][:local] = :time
+    options[:data].merge! find_format_data(format, :time)
 
-    time_tag time, time.strftime(format12), options
+    time_tag time, time.strftime(options[:data][:format]), options
   end
 
   def local_date(time, options = nil)
+    time = utc_time(time)
+
     options, format = extract_options_and_value(options, :format)
-    options[:format] = format || LocalTime.default_date_format
-    local_time time, options
+
+    options[:data] ||= {}
+    options[:data][:local] = :time
+    options[:data].merge! find_format_data(format, :date)
+
+    time_tag time, time.strftime(options[:data][:format]), options
```

Comment:
`#local_date` is _really_ just `#local_time` called with a date format. So it'd be great if we called `local_time` from here instead of `time_tag`. We should be able to do that if we determine the format before reaching this point and then passing it as a string to `#local_time`.